### PR TITLE
feat(content-sharing): Enable shared link deletion

### PR DESCRIPTION
**Pull request by sahiga**
This PR enables the deletion of shared links within the ContentSharing Element. (Follow-up to #2211)

### CODE REVIEW

File: src/elements/content-sharing/__tests__/ContentSharing.test.js [hunk 1]
```diff
@@ -373,6 +373,33 @@ describe('elements/content-sharing/ContentSharing', () => {
...
+            });
+            wrapper.update();
+            expect(share).toHaveBeenCalledWith(
+                { id: MOCK_ITEM_ID, permissions: {} },
+                ACCESS_NONE,
+                expect.anything(),
+                expect.anything(),
+                CONTENT_SHARING_SHARED_LINK_UPDATE_PARAMS,
+            );
+            expect(wrapper.find(UnifiedShareModal).prop('sharedLink')).toEqual(MOCK_NULL_SHARED_LINK);
```
Comment (mxiao6) [inline src/elements/content-sharing/__tests__/ContentSharing.test.js:401]: Can this be simplified to `usm.prop('sharedLink')`?

Comment (sahiga) [reply src/elements/content-sharing/__tests__/ContentSharing.test.js:401]: That was my initial thought as well. However, `usm` refers to the `UnifiedShareModal` in the wrapper _before_ the update, so we have to retrieve a new reference here.

File: src/elements/content-sharing/SharingNotification.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import API from '../../api';
 import Notification from '../../components/notification/Notification';
 import NotificationsWrapper from '../../components/notification/NotificationsWrapper';
 import { convertItemResponse } from '../../features/unified-share-modal/utils/convertData';
-import { ACCESS_COLLAB, STATUS_ERROR, TYPE_FILE, TYPE_FOLDER } from '../../constants';
+import { ACCESS_COLLAB, ACCESS_NONE, STATUS_ERROR, TYPE_FILE, TYPE_FOLDER } from '../../constants';
```
Comment (booc0mtaco) [inline src/elements/content-sharing/SharingNotification.js:9]: i am not sure what ACCESS_NONE means. does it actually correspond to a value returned by the API, or is it more of a placeholder when removing?

File: src/elements/content-sharing/ContentSharing.js [hunk 3]
```diff
@@ -159,7 +161,7 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
                         initialDataReceived
                         item={item}
                         onAddLink={onAddLink}
```
Comment (booc0mtaco) [inline src/elements/content-sharing/ContentSharing.js:163]: @il10 See here: 
- `onAddLink`
- `onRemoveLink`

File: src/elements/content-sharing/SharingNotification.js (hunk 2 above)
Comment (sahiga) [reply src/elements/content-sharing/SharingNotification.js:9]: Yup, it's a placeholder. `ShareAccessSelect` also uses `ACCESS_NONE`, so I retained that part of the Item `share()` code to ensure backwards compatibility.

**Pull request merged**